include/fmt/format.h

```diff
@@ -3484,9 +3484,18 @@ template <typename Char> struct udl_arg {
   }
 };
 
+// Converts string literals to basic_string_view.
 template <typename Char, size_t N>
-FMT_CONSTEXPR basic_string_view<Char> literal_to_view(const Char (&s)[N]) {
-  return {s, N - 1};
+FMT_CONSTEXPR basic_string_view<Char> fmt_string_to_view(const Char (&s)[N]) {
+  // Remove trailing null character if needed. Won't be present if this is used with
+  // raw character array (i.e. not defined as a string).
+  return {s, N - (std::char_traits<Char>::to_int_type(s[N - 1]) == 0 ? 1 : 0)};
+}
+
+// Converts string_view-like types to basic_string_view.
+template <typename Char, typename String>
+FMT_CONSTEXPR basic_string_view<Char> fmt_string_to_view(const String& s) {
```

Comment:
This needs to be more constrained. I suggest using `std_string_view` (an alias to `std::string_view`) here.

Reply:
It's not desirable that we constrain it to just std::string_view for us, since there are other compatible types outside the standard we use, such as folly::StringPiece, that std::string_view doesn't have implicit constructors for. Do you have any suggestion for how to better constrain it that retains use with compatible types?

Reply:
`folly::StringPiece` is morally obsolete and we should actively discourage new uses of it. Since this is a new API there is no reason to make it work with legacy types.